src/js/hooks/useStockListFilters.jsx

```diff
@@ -0,0 +1,111 @@
+import { useEffect, useState } from 'react';
+
+import queryString from 'query-string';
+import { useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+
+import filterFields from 'components/stock-list/FilterFields';
+import apiClient from 'utils/apiClient';
+import { transformFilterParams } from 'utils/list-utils';
+
+const useStockListFilters = () => {
+  const [filterParams, setFilterParams] = useState({});
+  const [defaultFilterValues, setDefaultFilterValues] = useState({});
+  const [filtersInitialized, setFiltersInitialized] = useState(false);
+
+  const [locations, setLocations] = useState([]);
+  const history = useHistory();
+
+  const { currentLocation } = useSelector(state => ({
+    currentLocation: state.session.currentLocation,
+  }));
+
+  const fetchLocationById = async (id) => {
+    const response = await apiClient(`/openboxes/api/locations/${id}`);
+    return response.data?.data;
```

Comment:
seems to be the same function that is used in purchase order list hook - that's why moving in to a util has sense